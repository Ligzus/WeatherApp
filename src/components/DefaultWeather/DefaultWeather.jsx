import { useEffect, useState } from "react";
import { getDefaultWeather } from "../../utils/weatherApi";
import { DEFAULT_CITIES } from "../../constants/cities";
import WeatherCard from "../WeatherCard/WeatherCard";
import styles from "./DefaultWeather.module.css";

function DefaultWeather() {
  const [weather, setWeather] = useState([]);
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchDefaultWeather = async () => {
      setLoading(true);
      try {
        const data = await getDefaultWeather();

        setWeather(
          DEFAULT_CITIES.map((city, index) => ({
            city,
            temp: data[index].main.temp,
            windSpeed: data[index].wind.speed,
            humidity: data[index].main.humidity,
            icon: data[index].weather[0].icon,
          }))
        );

        setError("");
      } catch (err) {
        setError(err.message || "Что-то сломалось, попробуй позже");
      } finally {
        setLoading(false);
      }
    };

    fetchDefaultWeather();
  }, []);

  return (
    <div className={styles.container}>
      <h2>Ты точно мечтаешь тут побывать:</h2>

      <div className={styles.cardContainer}>
        {loading && <p className={styles.loader}>загружаем погоду...</p>}
        {error && <p className={styles.error}>{error}</p>}
        {weather.length > 0 &&
          !loading &&
          weather.map((cityWeather, index) => (
            <WeatherCard key={index} weather={cityWeather} />
          ))}
      </div>
    </div>
  );
}

export default DefaultWeather;
