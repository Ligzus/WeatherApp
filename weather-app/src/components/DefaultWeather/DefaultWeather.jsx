import React, { useEffect, useState } from "react";
import { getDefaultWeather } from "../../utils/weatherApi";
import WeatherCard from "../WeatherCard/WeatherCard";
import styles from "./DefaultWeather.module.css";

function DefaultWeather() {
  const [weather, setWeather] = useState([]);
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchDefaultWeather = async () => {
      setLoading(true);
      try {
        const data = await getDefaultWeather();
        setWeather([
          {
            city: "Лондон",
            temp: data.firstCityData.main.temp,
            windSpeed: data.firstCityData.wind.speed,
            humidity: data.firstCityData.main.humidity,
            icon: data.firstCityData.weather[0].icon,
          },
          {
            city: "Париж",
            temp: data.secondCityData.main.temp,
            windSpeed: data.secondCityData.wind.speed,
            humidity: data.secondCityData.main.humidity,
            icon: data.secondCityData.weather[0].icon,
          },
          {
            city: "Саранск",
            temp: data.thirdCityData.main.temp,
            windSpeed: data.thirdCityData.wind.speed,
            humidity: data.thirdCityData.main.humidity,
            icon: data.thirdCityData.weather[0].icon,
          },
        ]);
        setError("");
      } catch (err) {
        setError("Что-то сломалось, попробуй позже");
      } finally {
        setLoading(false);
      }
    };

    fetchDefaultWeather();
  }, []);

  return (
    <div className={styles.container}>
      <h2>Ты точно мечтаешь тут побывать:</h2>
      {loading && <p className={styles.loader}>загружаем погоду...</p>}
      {error && <p className={styles.error}>{error}</p>}
      {weather.length > 0 && !loading && weather.map((cityWeather, index) => (
        <WeatherCard key={index} weather={cityWeather} />
      ))}
    </div>
  );
}

export default DefaultWeather;
