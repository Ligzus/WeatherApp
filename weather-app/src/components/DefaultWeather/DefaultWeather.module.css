.container {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  align-items: center;
  padding-left: calc(50% - 445px);
  padding-right: calc(50% - 445px);
}

.cardContainer {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.loader {
  text-align: center;
  font-size: 18px;
  color: #585858;
  margin-top: 25px;
}

h2 {
  margin: 0;
  padding: 35px 5px;
}

@media (min-width: 920px) {
  .container {
    align-items: flex-start;
  }

  .cardContainer {
    flex-direction: row;
    justify-content: center;
  }

  h2 {
    font-size: 32px;
  }
}

@media (max-width: 486px) {
  h2 {
    text-align: center;
  }
}
