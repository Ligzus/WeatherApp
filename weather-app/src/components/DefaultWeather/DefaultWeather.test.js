import React from "react"; // Добавь этот импорт
import { render, act } from "@testing-library/react";
import DefaultWeather from "./DefaultWeather";

test("renders DefaultWeather component correctly", async () => {
  await act(async () => {
    const { asFragment } = render(<DefaultWeather />);
    expect(asFragment()).toMatchSnapshot();
  });
});
