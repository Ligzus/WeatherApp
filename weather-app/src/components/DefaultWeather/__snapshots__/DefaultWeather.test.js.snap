// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DefaultWeather component correctly 1`] = `<DocumentFragment />`;
